### PR TITLE
Project: Members Only

### PR DESCRIPTION
**Pull request by t-yanick**
In this project, we :
- created a post model and controller
- Created a user model and controller
- Used devise gem for user authentication and factored our code such that only signed in members can know who made a particular post.

Thanks as we await feedback to improve the project.

### CODE REVIEW

File: app/views/posts/_feed.html.erb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+                            <% end %>
+                            <%= link_to edit_post_path(post), class: "level-item" do %>
+                            <span class="icon"><i class="fa fa-pencil" aria-hidden="true"></i></span>
+                            <% end %>
+                            <%= link_to post, method: :delete, data: { confirm: "Are you sure you want to delete this post?" } do %>
+                                    <span class="icon"><i class="fa fa-trash-o" aria-hidden="true"></i></span>
+                            <% end %>
+                        </div>
+                    </nav>
+                <% end %>
```
Comment (teekaytech) [inline app/views/posts/_feed.html.erb:46]: In the selected snippet above, you're having logic inside your view.
Kindly use the view helper to encapsulate this logic.

File: app/views/posts/_feed.html.erb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+    <% @posts.each do |post| %>
+        <div class="box">
+            <article class="media">
+                <div class="media-left">
+                <% if user_signed_in? %>
+                    <figure class="image is-64x64">
+                        <img src="https://bulma.io/images/placeholders/64x64.png">
+                    </figure>
+                    <% end %>
```
Comment (teekaytech) [inline app/views/posts/_feed.html.erb:21]: Here is another logic in the view.

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -1,15 +1,63 @@
...
+              </p>
+            <% else %>
+
+            <p class="control">
+              <%= link_to 'Sign In', new_user_session_path, class:"button is-info"%>
+            </p>
+            <p class="control">
+              <%= link_to 'Sign Up', new_user_registration_path, class:"button is-info"%>
+            </p>
+            <%end%>
```
Comment (teekaytech) [inline app/views/layouts/application.html.erb:56]: In the selected snippet above, you're having logic inside your view.
Kindly use the view helper to encapsulate this logic.

**Review by teekaytech — CHANGES_REQUESTED** [review]
## Status: Changes Required :warning: :vertical_traffic_light: 

### Hello team :wave:

__You've made laudable progress in this milestone.__

- Nice implementation fo flash messages for different actions :100: 
- Clean and organized codebase  :ok_hand: 
- Descriptive PR title and summary :+1: 

_However, some changes are required before this project can be approved. Please check my comments below and fix them as appropriate._

**[ REQUIRED CHANGES ]**

- [ ] You should create the post routes using `resources :posts, only: [:new, :create, :index]`  ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_routes_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))

- [ ] Files not related to the project should not be committed and added to .gitignore (e.g. .DS_Store) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/gitignore.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))

- [ ] View-related logic can be encapsulated in View helpers (app/helpers/\*_helper.rb)
- Kindly check the highlighted inline comments below for instances where you've added logic directly inside the views and encapsulate them in view helpers. 
- [This article](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers) might help in doing so.


**[ OPTIONAL CHANGES ]**

- [ ] You added some comments about the code in your `posts_controller.rb`. I personally believe these comments are unnecessary. This is because the actions' names are descriptive enough to know what they do and how to treat them.

- [ ] Also, for your `create`, `update`, and `destroy` actions in the `posts_controller.rb` scripts, you specified both HTML and JSON formats for the output. I think the JSON output format is not needed, since we're not building an API. 
* * *
**Please make the required changes and submit another review request when you are done.**

* * *

**Taofeek Olalere
[Twitter](https://twitter.com/ola_lere) | [Github](https://github.com/teekaytech) | @taofeekolalere on Slack**
__Happy Coding :smiley:__

**Review by rOluochKe — APPROVED** [review]
Hello Team
### Well done, you got it right :clap: The project has been approved. Congratulations :boom: :100:
### Merge and submit the project. Good luck with the next one.
- Happy coding!

**Pull request merged**